### ISSUE TITLE
Make pry plugins configurable

### ISSUE THREAD

**Issue by aried3r** [plribeiro3000/jazz_fingers#9]
Currently, all pry plugins are required. In my case, I don't need some, for example pry-git, which holds the mime-types gem hostage. It would be great to be able to configure only those you need.

Ironically, this was somewhat possible while [this bug](https://github.com/plribeiro3000/jazz_fingers/issues/5#issuecomment-112857970) was present ;)

**Comment by plribeiro3000** [comment]
@aried3r Gotcha. While what you are saying make sense i believe it would break the idea of this repo which is to be a gem to add everything for you without worries but at the same time i do agree that some of the dependencies are a pain in the ass because of bugs, just like one i encountered some time ago with `yard` and `ENV` variables.

So, with all this context, do you have a solution in mind? I believe we could work towards something like the `jazz_fingers/setup` file. What do you think?

**Comment by illtellyoulater** [comment]
Also, at the moment pry-coolline fails to work on Windows ( https://github.com/plribeiro3000/jazz_fingers/issues/10 ) presumably because of a issues in coolline ( https://github.com/Mon-Ouie/coolline/issues/19 ).

So I had to uninstall jazz_fingers to be able to use pry again :(
If it was possible to choose which plugin to (not) load then jazz_fingers could be a bit more stable in these cases when plugins break.

**Comment by plribeiro3000** [comment]
`jazz_fingers` is just a thin layer on top of several other gems. If we have a problem with the dependencies i'm afraid that making them configurable sounds as a workaround instead of a fix.

Having this in mind i think that the only issues that need to be addressed in this repo are related to conflicts between those dependencies instead of trying to solve those issues in this repository.

As to be able to use `jazz_fingers` without those issues i think that one possible solution is to be pessimistic on the dependencies and avoid broken ones. Or not not have fixed dependencies at all in a way that the final user can easily fix any problem by forcing a specific version. What do you think?

**Comment by aried3r** [comment]
> jazz_fingers is just a thin layer on top of several other gems. If we have a problem with the dependencies i'm afraid that making them configurable sounds as a workaround instead of a fix.

It's just that the dependencies are old. [pry-git](https://github.com/pry/pry-git) hasn't been updated since 4 years ago which itself depends on [grit](https://github.com/mojombo/grit) which isn't maintained anymore. Maybe make the very old dependencies, beginning with pry-git optional? It's mostly this one which holds dependencies "hostage" in my case.

I'll have a closer look at the code of this repo what exactly it does and if I need the glue. I really enjoy this projects, but depending on your decision regarding dependencies, I'll probably fork or cook my own. This is not a threat, I really do appreciate your work on reviving jazz_hands! :)

That being said, I don't know enough about gem development but from what I've gathered, true optional dependencies aren't possible. Usually just using something like

``` Ruby
begin
  require 'a'
rescue LoadError
  # The 'a' gem is not installed
  require 'b'
end
```

**Comment by plribeiro3000** [comment]
Got it. I think we should just get rid of `pry-git` then.

As for the dependency, is really hard. The problem is [here](https://github.com/plribeiro3000/jazz_fingers/blob/master/jazz_fingers.gemspec#L21-L29).

If we do remove it from the `gemspec` then the user will have to load everything by hand and thus removing the need of this project. Maybe we can work on the dependencies and keep only the active and working ones. What do you think?

**Comment by aried3r** [comment]
Keeping only the active ones sounds good as a first step. :)

Also: https://bibwild.wordpress.com/2015/09/09/optional-gem-dependencies

**Comment by plribeiro3000** [comment]
Alright. Do you mind to send a PR for that @aried3r?

Otherwise i will do it myself but might take some time to do it.

**Comment by aried3r** [comment]
Will try, taking a quick glimpse, I couldn't find anything relating to pry-git except for the gem spec file and https://github.com/plribeiro3000/jazz_fingers/blob/master/lib/jazz_fingers.rb#L7, so removing those should do it, right?

**Comment by plribeiro3000** [comment]
Yep. Exactly. =)

**Comment by plribeiro3000** [comment]
Closing this since #12 was already merged and released as in `3.0.0`.

Thanks for your work here @aried3r !
